[9.0][mail_activity][mail_activity_calendar][mail_activity_crm]

JordiBForgeFlow opened the pull request:
Implements the activities of 11.0 to 9.0 (and later 10.0), with some minor simplifications.

This PR includes the following modules:
* mail_activity
* mail_activity_calendar
* mail_activity_crm

Activities
==========

This module backports the activities from 11.0.

Known issues / Roadmap
======================

* The button 'Mark done' that appears for each activity in the chatter
  is not available.

* It is not possible to schedule a next activity based on a given activity

Mail Activity Calendar
======================

Adds the ability to create calendar meetings when you schedule an
activity of category 'Meeting'.

Known issues / Roadmap
======================

* When the user is redirected from the activity to the calendar view,
  the summary of the activity does not appear in the pop-up window following
  the creation of a new calendar event.

Activities in CRM
=================

This module adds activities to CRM.

Known issues / Roadmap
======================

* From the Kanban view you can only display the activities. You cannot
  schedule new activities from the kanban pop-up window, and the layout is
  slightly different.

* The Kanban view shows the next activity date and summary, in red if it
  is overdue.

hbrunn committed:
[ADD] mail_activity

hbrunn committed:
fixup! [ADD] mail_activity

hbrunn committed:
[ADD] inject xmlids in an indirect way to avoid warnings

hbrunn committed:
[ADD] mail_activity_type_id for mail.message

hbrunn committed:
fixup! [ADD] inject xmlids in an indirect way to avoid warnings

hbrunn committed:
[ADD] (part of) v11 UI

andreparames committed:
[PORT][9.0] mail_activity

This a port to v9 of the port done in PR #204. The only significant
change is the method _setup_complete(), which had to be adapted to
fix a bug in Odoo (already corrected in v10 and above).

andreparames committed:
[ADD][9.0] setup/mail_activity

andreparames committed:
[9.0] mail_activity: convert mixin to decorator

The Odoo v9 model mechanism has some issues with loading
AbstractModels as mixins; specifically, the computed fields of the
original model no longer work reliably.

This new approach uses a Python decorator to change the class when
the module is being loaded, so the Odoo model mechanisms only see
a class that does a simple inheritance, and already "comes" with
the new fields/methods related to the activities.

It's essentially the same as copy-pasting the fields and methods
directly into the class.

ThomasBinsfeld committed:
[ADD] Mail Activity: french translations from Odoo 11.0

JordiBForgeFlow committed:
9.0 add mail_activity, mail_activity_crm, mail_activity_calendar

tarteo commented on mail_activity/views/mail_activity.xml:
```diff
@@ -0,0 +1,181 @@
+                        <field name="previous_activity_type_id"/>
+                        <field name="has_recommended_activities"/>
+                    </group>
+                    <group attrs="{'invisible': [('has_recommended_activities','=',False)]}">
```
I've set 1 recommended next activity on a type:

![image](https://user-images.githubusercontent.com/10028499/41642279-27d8f692-7468-11e8-8dce-3ddf22b6df93.png)

When I select the activity type:

![image](https://user-images.githubusercontent.com/10028499/41642348-4e53437c-7468-11e8-8ed2-ecb07b2d28b3.png)

The radio button doesn't seem to appear.

JordiBForgeFlow replied:
I have not fixed. But updated the README stating as known issue.

tarteo commented on mail_activity/static/src/less/systray.less:
```diff
@@ -0,0 +1,41 @@
+// Navbar icon and dropdown
+.o_mail_navbar_item {
```
Can you improve the layout of the activitymenu items, please?
![image](https://user-images.githubusercontent.com/10028499/41641749-9baaef64-7466-11e8-9b06-642c4e573d7a.png)

JordiBForgeFlow replied:
Looks better now :)

![image](https://user-images.githubusercontent.com/7683926/41669747-d3216436-74b2-11e8-9808-72e75dfad251.png)

tarteo commented on mail_activity/static/src/js/systray.js:
```diff
@@ -0,0 +1,148 @@
+     * @param {Boolean} [data.activity_created] when activity created
+     */
+    _updateCounter: function (data) {
+        if (data) {
```
You can also do `if(!data) { return; }` here. Makes the code flatter, but it's nitpicking so not really important :smile:

JordiBForgeFlow replied:
I'd prefer the code as similar to the original as possible

tarteo reviewed (changes requested):
Thanks for the back port! I reviewed the module code and functionality and have some small comments.

JordiBForgeFlow commented:
@tarteo I have added mail_activity_calendar and mail_activity_crm. With the 3 modules the backport of activities to 9.0 and 10.0 will be much more complete!

fuentes73 commented:
Thanks, great job!  
Testing the module I've seen that if you schedule a call to a partner horizontal scroll is added to the view

![captura de pantalla 2018-06-21 a las 10 21 03](https://user-images.githubusercontent.com/4355395/41706884-2bda1700-753d-11e8-994f-5eca21be2a92.png)

A solution could be:
`.o_avatar_stack {position: relative}`

Plus, the action icon and it's border are set in white color. It should be in another color more visible,  isn't it?
![captura de pantalla 2018-06-21 a las 10 31 57](https://user-images.githubusercontent.com/4355395/41707499-87cb55d2-753e-11e8-84dc-5b4669dd488a.png)
 
![captura de pantalla 2018-06-21 a las 10 41 59](https://user-images.githubusercontent.com/4355395/41707999-c7573670-753f-11e8-8a47-13e71ccb0635.png)

JordiBForgeFlow commented:
@fuentes010 I am not an expert in frontend development. Perhaps you know better and can make some quick adjustments? Please send me a PR with your proposed fixed, and hopefully we can fix this together.

tarteo reviewed (changes requested):
@jbeficent Thanks! The calendar feature is really nice. I get an error though when I try to cancel the activity in the chatter.
`
Uncaught TypeError: self._rpc is not a function
http://3329956-286-b08677.runbot1.odoo-community.org/mail_activity_calendar/static/src/js/mail_activity.js:44
Traceback:
TypeError: self._rpc is not a function
    at Class.confirm_callback (http://3329956-286-b08677.runbot1.odoo-community.org/mail_activity_calendar/static/src/js/mail_activity.js:44:37)
    at HTMLButtonElement.<anonymous> (http://3329956-286-b08677.runbot1.odoo-community.org/web/static/src/js/framework/dialog.js:87:41)
    at HTMLButtonElement.dispatch (http://3329956-286-b08677.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:4641:58)
    at HTMLButtonElement.elemData.handle (http://3329956-286-b08677.runbot1.odoo-community.org/web/static/lib/jquery/jquery.js:4309:63)
`

I think instead of using `_rpc` you can use `rpc`. I think _rpc is introduced in 11.0 in the ServicesMixin in 9.0 the function is directly on web.Widget

tarteo commented:
@jbeficent Can I create a PR on your fork / branch to fix the cancellation issue?

JordiBForgeFlow commented:
Sure!

El vie., 22 jun. 2018 19:10, Dennis Sluijk <notifications@github.com>
escribió:

> @jbeficent <https://github.com/jbeficent> Can I create a PR on your fork
> / branch to fix the cancellation issue?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/social/pull/286#issuecomment-399514259>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHU_VvQ7NxEafAweh7EUjUr4t7ZdbILYks5t_SUZgaJpZM4UtO1g>
> .
>

fuentes73 committed:
[FIX] mail_activity: avatar icon style

fuentes73 commented:
@jbeficent I've already done the [PR](https://github.com/Eficent/social/pull/1)

JordiBForgeFlow commented:
Thanks Pablo!

El sáb., 23 jun. 2018 12:58, Pablo Fuentes <notifications@github.com>
escribió:

> @jbeficent <https://github.com/jbeficent> I've already done the PR
> <https://github.com/Eficent/social/pull/1>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/social/pull/286#issuecomment-399665664>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AHU_VikNTsRB0nzBc5o_GsEanRb5A_9Bks5t_h9ggaJpZM4UtO1g>
> .
>

tarteo committed:
[FIX] Canceling activity with meeting attached

tarteo commented:
@jbeficent Made the fix https://github.com/Eficent/social/pull/2

hbrunn reviewed (dismissed):
please fix https://travis-ci.org/OCA/social/jobs/396943416#L576

JordiBForgeFlow commented:
@hbrunn We're good to go. Can you please merge?

hbrunn commented:
can we have more test coverage?

JordiBForgeFlow commented:
@hbrunn This would mean adding JS tests, which I am totally unable to add.. And adding python tests does not either make much sense, since the data model is pretty simple.

JordiBForgeFlow committed:
[add] mail_activity_calendar: init hook to poperly set up calendar.event model

odony commented on mail_activity/README.rst:
```diff
@@ -0,0 +1,65 @@
+Contributors
+------------
+
+**This module is a backport from Odoo SA and as such, it is not included in the OCA CLA. That means we do not have a copy of the copyright on it like all other OCA modules.**
```
Guys, do you have guidelines somewhere related to porting code from Odoo SA (or other third-party code)? The GPL licenses require that you identify specifically in each file what code is covered by what copyright and what license (when not uniform).

I find the current ports a little confusing, with a mix of licensing and copyright info. I see no copyright notice from Odoo SA (and its third-party contributors), and I can't tell by reading the source files who owns the copyright on what part of the code, nor what you changed (perhaps I simply missed it?)

For the record, the short notice "# Part of Odoo ..." only works inside the official Odoo code because:
 - it comes with matching LICENSE and COPYRIGHT files
 - external contributions are always subject to a CLA and authorship attribution is done via Git metadata on commits.
When you copy the code outside of official branches you can't replicate all of these conditions AFAICT, so you need to provide proper attribution, license and copyright info in each file, in context, and identify all your changes.

I know this stuff can be a PITA, but it's a key principle of open source, and I know we all want to do it right :-)

pedrobaeza replied:
@odony AFAIK, the module is full licensed as LGPL-3, which matches your license, isn't it?

odony replied:
@pedrobaeza well, there seems to be some conflicting info about that too, even in the manifest file ;-)
https://github.com/Eficent/social/blob/c3c1af5a6c02f80cad5158d3f6914311eebeb96b/mail_activity/__openerp__.py#L3

JordiBForgeFlow replied:
@odony I totally agree with you. I will add Odoo, S.A. wherever the code was moved, and change the licensing type. This feature is super useful, and unfortunately not all companies are able to move to 11.0 as quick as we'd like, so backporting is sometimes an option.

pedrobaeza replied:
You are right, @odony. Some copy/paste error, but @jbeficent is on it as he has said. Thanks for noticing.

JordiBForgeFlow committed:
update copyright

JordiBForgeFlow committed:
fixup!

JordiBForgeFlow commented:
@odony I have added Odoo, S.A. explicitly in the copyrights section in all files. 
As for identifying the changes, the readme file in each module identifies, in the section 'Known issues / Roadmap' the differences with respect to 11.0. It is hard to identify differences explicitly in code. Code had do be adjusted quite a lot in the JS side, and in python code was extracted in bits and pieces from odoo 11.0, because in the core it was entangled with other  non activity-specific features.

odony commented:
Thanks for updating the copyright mentions!


> As for identifying the changes, the readme file in each module identifies, in
> the section 'Known issues / Roadmap' the differences with respect to 11.0. It
> is hard to identify differences explicitly in code.

Sure, and if every file contains up-to-date mentions about the licenses and
copyright involved (or points to LICENSE/COPYRIGHT files that do), you have a
very defensible position that the license requirements are satisfied :)

Revision history can then be used to more precisely analyze ownership of code
lines. Especially if you have an initial commit to check in the original code,
and then build over that with separate commits (this could be part of some
guidelines)

JordiBForgeFlow commented:
@odony In cases where Odoo discontinues a module, and someone wants to continue the work preserving commit history should be mandatory, and even simple using a standard code migration such as the ones OCA uses when migrating modules across major versions https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-11.0. 


In this specific case this was not possible because we are not backporting an entire module, but specific features contained in various other modules. So preserving commit history seems impossible.

Thanks for your feedback anyway!

rafaelbn commented:
Hi @hbrunn , i think you review is solved :smile: :+1:

JordiBForgeFlow committed:
mail_activity: When an activity is complete, it should be logged
with the correct subtype. This fix won't be required in v10.

JordiBForgeFlow commented:
@hbrunn please notice 6e0429c. If you use some variation of mail_activity this is important to take into account.

JordiBForgeFlow committed:
show expected closing date

JordiBForgeFlow committed:
- mail_activity:allow to set an activity of category meeting to done.
- mail_activity_crm: hide old crm.activity menus, add migration scripts
  from crm.activity.

agb80 commented:
Does this MR suppressed this other one: https://github.com/OCA/social/pull/286?

paulius-sladkevicius commented:
Great initiative! Looking forward to test it out on v10, so if only @hbrunn is holding it to have merged and move to have it ready for v10?

hbrunn commented:
oh, it's fivethousand years ago I reviewed this and I don't have time for proper review now, so I dismissed my review

hbrunn commented:
PS: I'd still refuse to merge because it fails pylint currently...

inspiredbusiness commented:
Hi,
Not sure if this was the best way to go about this, but we needed this in v10. There were issues merging this branch into v10 so that I could work on the port. I ended up copy/paste the files in this branch into a new v10 branch, committed, then did the work to port to v10. Let me know if there is anything I can do to get traction on this.

I have submitted a PR for this in v10

https://github.com/OCA/social/pull/328

inspiredbusiness commented:
I have made changes in https://github.com/OCA/social/pull/328 that allows this to work with the crm activities report. Just had to add a hook to copy from mail compose to mail message and add the missing field to the crm report sql view. Like v12 the activity will not show in the activity report until it is completed and converted into a mail message

LoisRForgeFlow commented:
@inspiredbusiness for v10 there was this PR: https://github.com/OCA/social/pull/204

inspiredbusiness commented:
@lreficent I saw that PR but the first comment said it needed a UI to be useful. The branches were confusing and I could not get then to merge correctly and the v9 version seemed more complete. Feel free to ignore my PR. It seems all of the branches for this are stuck and I had a customer that could not wait any longer. So I ported v9 which seemed to be the most complete and have gone to production with the PR + a few other minor changes I never ended up pushing yet to GH. Let me know if there is anything I can do to make all of this get moving again.

LoisRForgeFlow commented:
@inspiredbusiness Thank for the insights, from my side I just wanted to make you aware of the existence of another PR, maybe contributors of that other PR could give you more instructions on how to converge the work to a common point.
